src/components/widget.js

```diff
@@ -227,80 +343,124 @@ class Widget extends LitElement {
+      await this._loadMemberData();
+      await this._loadAdditionalData();
 
-      const { status, success, response } = await chrome.runtime.sendMessage({
-        operation: "LOAD_MEMBER_DATA",
-        username: this.username,
-        platform: this.platform,
-        ORBIT_CREDENTIALS,
-      });
+      this.isLoading = false;
+      this.hasLoaded = true;
+      this.requestUpdate();
+    }
+  }
 
-      this.workspace = ORBIT_CREDENTIALS.WORKSPACE;
+  /**
+   * Fetch member data & store it in state
+   */
+  async _loadMemberData() {
+    const { status, success, response } = await chrome.runtime.sendMessage({
+      operation: "LOAD_MEMBER_DATA",
+      username: this.username,
+      platform: this.platform,
+    });
 
-      if (status === 401) {
-        this.hasAuthError = true;
-      } else if (status === 404) {
+    if (status === 401) {
+      this.hasAuthError = true;
+    } else if (status === 404) {
+      this.isAMember = false;
+      this.workspace = response?.workspace;
+    } else if (success === false) {
+      this.hasError = true;
+    } else {
+      const { data, included, workspace } = response;
+
+      if (!data) {
         this.isAMember = false;
-      } else if (success === false) {
         this.hasError = true;
-      } else {
-        const { data, included } = response;
-
-        if (!data) {
-          this.isAMember = false;
-          this.hasError = true;
-          this.isLoading = false;
-
-          this.requestUpdate();
-          return;
-        }
-
-        this.isAMember = true;
-
-        const identities = data.relationships.identities.data.map(
-          ({ id, type }) =>
-            included.find(
-              ({ id: included_id, type: included_type }) =>
-                id === included_id && type === included_type
-            )?.attributes
-        );
-
-        const organizations = data.relationships.organizations.data.map(
-          ({ id, type }) =>
-            included.find(
-              ({ id: included_id, type: included_type }) =>
-                id === included_id && type === included_type
-            )?.attributes
-        );
-
-        const organization = organizations[0] || null;
-
-        this.member = {
-          name: data.attributes.name,
-          jobTitle: data.attributes.title,
-          slug: data.attributes.slug,
-          teammate: data.attributes.teammate,
-          orbitLevel: data.attributes.orbit_level,
-          organization: organization,
-          lastActivityOccurredAt: data.attributes.last_activity_occurred_at,
-          tags: data.attributes.tags,
-          identities: identities,
-        };
+        return;
       }
 
+      this.workspace = workspace;
+      this.isAMember = true;
+      this.member = buildMemberData(data, included);
+    }
+  }
+
+  /**
+   * Fetch additional data & store it in state
+   */
+  async _loadAdditionalData() {
+    if (this.platform !== "github") return;
```

Comment:
I think we can implement this `return` in `background.js` directly—so that `widget.js` does not have special cases for specific platforms

Reply:
Note: this can be moved out of this PR and into the future effort to make `background.js` smarter and `widget.js` simpler—it might even be better to do so all in one PR dedicated to that